lib/SPIRV/SPIRVReader.cpp

```diff
@@ -475,6 +475,12 @@ Type *SPIRVToLLVM::transType(SPIRVType *T, bool IsClassMember) {
         getSPIRVTypeName(kSPIRVTypeName::JointMatrixINTEL, SS.str());
     return mapType(T, getOrCreateOpaquePtrType(M, Name));
   }
+  case OpTypeForwardPointer: {
+    SPIRVTypeForwardPointer *FP =
+        static_cast<SPIRVTypeForwardPointer *>(static_cast<SPIRVEntry *>(T));
+    return mapType(T, transType(static_cast<SPIRVType *>(
+                          BM->getEntry(FP->getPointerId()))));
```

Comment:
What about
```suggestion
                          BM->getEntry(FP->getOperand(0)->getId()()))));
```
to not add extra class member?

Reply:
@AlexeySachkov is right, we can't access the Pointer field here as it's not known yet.